src/Agent.Worker/JobRunner.cs

```diff
@@ -65,21 +65,24 @@ public sealed class JobRunner : AgentService, IJobRunner
                 jobContext.Variables.Set(Constants.Variables.Agent.WorkFolder, IOUtil.GetWorkPath(HostContext));
                 jobContext.Variables.Set(Constants.Variables.System.WorkFolder, IOUtil.GetWorkPath(HostContext));
 
-                // prefer task definitions url, then TFS url
+                // prefer task definitions url, then TFS collection url, then TFS account url
                 var taskServer = HostContext.GetService<ITaskServer>();
-                string taskUrl = jobContext.Variables.System_TaskDefinitionsUri;
                 Uri taskServerUri;
-                if (string.IsNullOrEmpty(taskUrl))
+                if (!string.IsNullOrEmpty(jobContext.Variables.System_TaskDefinitionsUri))
                 {
-                    taskServerUri = ReplaceWithConfigUriBase(message.Environment.SystemConnection.Url);
-                    Trace.Info($"Creating task server with tfs server url {taskServerUri.ToString()}");
+                    taskServerUri = ReplaceWithConfigUriBase(new Uri(jobContext.Variables.System_TaskDefinitionsUri));
+                }
+                else if (!string.IsNullOrEmpty(jobContext.Variables.System_TFCollectionUrl))
+                {
+                    taskServerUri = ReplaceWithConfigUriBase(new Uri(jobContext.Variables.System_TaskDefinitionsUri));
                 }
                 else
                 {
-                    taskServerUri = ReplaceWithConfigUriBase(new Uri(taskUrl));
-                    Trace.Info($"Creating task server with {taskServerUri.ToString()}");
+                    var configStore = HostContext.GetService<IConfigurationStore>();
+                    taskServerUri = ReplaceWithConfigUriBase(new Uri(configStore.GetSettings().ServerUrl));
```

Comment:
Did you mean to switch the last fallback to use the config setting URL? Previously the last fallback was message.Environment.SystemConnection.Url. Is this similar to the changes Stan made recently regarding the `.local` issue on OSX/Linux?